ports/esp32/machine_touchpad.c

```diff
@@ -29,12 +29,16 @@
 #include "modmachine.h"
 #include "driver/gpio.h"
 
-#if CONFIG_IDF_TARGET_ESP32 || CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3
+#if SOC_TOUCH_SENSOR_SUPPORTED
 
 #if CONFIG_IDF_TARGET_ESP32
 #include "driver/touch_pad.h"
-#elif CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3
+#define TOUCH_HW_VER 1
+#elif SOC_TOUCH_VERSION_2
 #include "driver/touch_sensor.h"
+#define TOUCH_HW_VER 2
```

Comment:
Could the code below just use the IDF `SOC_TOUCH_VERSION_x` macros instead of defining a custom one?

And maybe to be sure the logic works, `#define SOC_TOUCH_VERSION_1 (0)` when `SOC_TOUCH_VERSION_2` is defined/enabled, and vice versa?

See also comments below.

Reply:
Sure. I think I looked for `SOC_TOUCH_VERSION_1` and thought it was never defined in ESP-IDF (only V2). Although I looked again now and it is defined. Will change.